File: chap05/run-model5-4.R
library(cmdstanr)

d <- read.csv(file='input/data-shopping-2.csv')
d$Income <- d$Income/100
data <- list(N=nrow(d), Sex=d$Sex, Income=d$Income, M=d$M, Y=d$Y)

model <- cmdstan_model(stan_file='model/model5-4.stan')
fit <- model$sample(data=data, seed=123, parallel_chains=4)
fit$save_object(file='output/result-model5-4.RDS')

File: chap05/run-model5-4.py
import pandas
import cmdstanpy

d = pandas.read_csv('input/data-shopping-2.csv')
d.Income /= 100
data = d.to_dict('list')
data.update({'N':len(d)})

model = cmdstanpy.CmdStanModel(stan_file='model/model5-4.stan')
fit = model.sample(data=data, seed=123, parallel_chains=4)
fit.save_csvfiles('output/result-model5-4')
